flink-ml-core/src/main/java/org/apache/flink/ml/param/ParamValidators.java

```diff
@@ -100,4 +100,22 @@ public boolean validate(T value) {
     public static <T> ParamValidator<T[]> nonEmptyArray() {
         return value -> value != null && value.length > 0;
     }
+
+    // Check if every element in the array-typed parameter value is in the array of allowed values.
+    public static <T> ParamValidator<T[]> isSubArray(T... allowed) {
```

Comment:
nit: `subArray` seems a bit confusing here since `subArray` indicates that the order should also be the same with the `allowed`, which is not true. How about `isSubSet`?